pypeit/reduce.py

```diff
@@ -472,9 +472,19 @@ def run_extraction(self, global_sky, sobjs_obj, skymask, ra=None, dec=None, obst
             if self.par['flexure']['spec_method'] != 'skip' and not self.std_redux:
                 self.spec_flexure_correct(mode='global')
 
-            # Extract + Return
-            self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs \
-                = self.extract(self.global_sky, self.sobjs_obj)
+            if self.par['reduce']['extraction']['skip_extraction']:
```

Comment:
I think this can be simplified to have line 469 state: 

469 if self.nobj > 0 and not self.par['reduce']['extraction']['skip_extraction']:

490 else: 

You are doing the same setups twice in 476-483 that are being done in the else. 

Ideally we would not have entered run_extraction at all, which was my initial intent, however I can see here that this routine is creating a bunch of outputs like tilts, waving etc. that are needed.  I'm pretty sure this is massively inefficient, since those images were probably already created to create the global sky. That is why this class needs to be refactored, which is on my list of things to do.